client/uma_token_source.go

```diff
@@ -0,0 +1,162 @@
+package client
+
+import (
+	b64 "encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/bitrise-io/bitrise-oauth/config"
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/clientcredentials"
+)
+
+const (
+	umaGrantType     = "urn:ietf:params:oauth:grant-type:uma-ticket"
+	claimTokenFormat = "urn:ietf:params:oauth:token-type:jwt"
+)
+
+type tokenJSON struct {
+	AccessToken  string `json:"access_token"`
+	TokenType    string `json:"token_type"`
+	RefreshToken string `json:"refresh_token"`
+	ExpiresIn    int32  `json:"expires_in"`
+}
+
+func (e *tokenJSON) expiry() (t time.Time) {
+	if v := e.ExpiresIn; v != 0 {
+		return time.Now().Add(time.Duration(v) * time.Second)
+	}
+	return
+}
+
+// UMATokenSource returns a token source that returns a new token each time the Token()
+// method is called.
+type UMATokenSource interface {
+	Token(claim interface{}, permisson []Permission, audienceConfig config.AudienceConfig) (*oauth2.Token, error)
+}
+
+type umaTokenSource struct {
+	config clientcredentials.Config
+}
+
+// NewUMATokenSource returns a new UMA token source.
+func NewUMATokenSource(config clientcredentials.Config) UMATokenSource {
```

Comment:
I don't think we need this function to be exposed since `func (cws *WithSecret) UMATokenSource() UMATokenSource` should done this.

Please also change the return type to `umaTokenSource`.